<?xml version="1.0"?>
<launch>

    <!-- <arg name="start_pt" default= '-7.43, -15.3, 0.8978652, 0.0'/> -->
    <!-- <arg name="start_pt" default= '-7.43, -15.3, 0.0, 0.0'/> -->
    <arg name="start_pt" default= '14.47, 1.6, 0.0, 0.0'/>
    <arg name="use_rviz" default="false"/>
    <arg name="use_matplotlib" default="false"/>
    <arg name="is_sim" default="false"/>
    <!-- <arg name="map_file" default="$(find svea_core)/resources/maps/floor2.yaml" /> -->
    <arg name="map_file" default="$(find svea_core)/resources/maps/gimp/floor2_rot.yaml" />
    <arg name="run_lidar" default="true"/>
    <arg name="use_mpc" default="true"/>

    <group if="$(eval use_rviz == true)">
        <node type="rviz" name="rviz" pkg="rviz" args="-d $(find svea_core)/resources/rviz/SVEA_floor2.rviz" />
    </group>

    <!-- <node pkg="map_server" type="map_server" name="map_server" output="screen" args="$(arg map_file)" /> -->
    <rosparam command="load" file="$(find svea_core)/resources/params/obstacles_rot.yaml" />

    <include file="$(find svea_sensors)/launch/localize.launch">
        <arg name="use_rs" value="true"/>
        <arg name="file_name" value="$(arg map_file)"/>
    </include>

    <!--open serial connection for controlling SVEA-->
    <node pkg="rosserial_python" type="serial_node.py" name="serial_node">
        <param name="port" value="/dev/ttyACM0"/>
        <param name="baud" value="250000"/>
    </node>

    <group if="$(eval use_mpc == false)">
    <node pkg="svea_core" type="floor2_team1.py" name="floor2_team1" output="screen">
        <param name="start_pt" value="$(arg start_pt)"/>
        <param name="use_rviz" value="$(arg use_rviz)"/>
        <param name="use_matplotlib" value="$(arg use_matplotlib)"/>
        <param name="is_sim" value="$(arg is_sim)"/>
        <param name="run_lidar" value="$(arg run_lidar)"/>
    </node>
    </group>

    <group if="$(eval use_mpc == true)">
    <node pkg="svea_core" type="floor2_mpc_co_av.py" name="floor2_team1" output="screen">
        <param name="start_pt" value="$(arg start_pt)"/>
        <param name="use_rviz" value="$(arg use_rviz)"/>
        <param name="use_matplotlib" value="$(arg use_matplotlib)"/>
        <param name="is_sim" value="$(arg is_sim)"/>
        <param name="run_lidar" value="$(arg run_lidar)"/>
    </node>
    </group>

    <include file="$(find svea_core)/launch/costmap_2d.launch">
    </include>

  	<!-- <include file="$(find svea_core)/launch/hector_mapping.launch"> --> 
        <!-- <remap from="map" to="explored_map" /> -->
        <!-- <arg name="tf_map_scanmatch_transform_frame_name" default="scanmatcher_frame"/> -->
        <!-- <arg name="base_frame" default="base_link"/> -->
        <!-- <arg name="odom_frame" default="odom"/> -->
        <!-- <arg name="pub_map_odom_transform" default="false"/> -->
        <!-- <arg name="scan_subscriber_queue_size" default="5"/> -->
        <!-- <arg name="scan_topic" default="scan"/> -->
        <!-- <arg name="map_size" default="1024"/> -->
    <!-- </include> -->

    <node pkg="svea_core" type="map_explore_node.py" name="explore_node" output="screen" />
    <node pkg="svea_core" type="collision_node.py" name="collision_node" output="screen" />
    <node pkg="svea_core" type="global_path_node.py" name="global_path_node" output="screen" />    


</launch>
